airflow/providers/sftp/sensors/sftp.py

```diff
@@ -71,6 +71,8 @@ def poke(self, context: 'Context') -> bool:
                 self.actual_file_to_check = file_from_pattern
             else:
                 return False
+        else:
+            self.actual_file_to_check = self.path
```

Comment:
Good catch! IMO it's probably not worth having `actual_file_to_check` be an instance attr since it's not used outside of the `poke()` method. WDYT? If you agree, it should be deleted from `__init__()` as well.

Reply:
Maybe we want to access one of the filenames that matches `self.file_pattern`, when `self.file_pattern` is not empty? I am not sure about that. We probably need some input from the committer of file_pattern supporting.
My true opinion is that the implementation about file_pattern is not very elegant. The variable `self.path` means a **file** path when `self.file_pattern` is empty. But when `self.file_pattern` is specified, the variable `self.path` is treated as a **folder** path.

Reply:
Indeed a good catch! :) 
I agree that `actual_file_to_check` can be deleted from the `__init__()` method, it's not really needed there, and it was my mistake to leave it there 😅 
Perhaps the best approach is to initiate the variable inside the `poke()` method itself.
Or, because it is a user-facing attribute, maybe it's ok to add this to the `template_fields`? WDYT? 

About the API itself, when I've added this feature, I saw `self.path` variable to be a strict and complete file location, but when you ask for a pattern, it's "easier" to understand that you look for a pattern, in a specific path.
So for example, if you want to check if csv reports are done and placed in the correct folder, you will give the folder with the `*.csv` pattern for poking.

But if you think that this API is not good, I guess it's ok also to change it :)

Reply:
Personally, I think it would be very cool if I could use sensors like this:
```
sensor_task = SFTPSensor(task_id='task_id', sftp_conn_id='sftp_conn_id', path='data/folder/datafile_{{ds}}_part_?.[ct]sv')
```
Also, this supporting should be implemented on some base class, so that all the file sensors could be supported.

Reply:
❤️ the discussion here! I think they are all great suggestions, but they don't necessarily need to be part of this PR. Definitely feel free to fire over PRs and we can discuss more.

>Perhaps the best approach is to initiate the variable inside the poke() method itself.
+1

Reply:
Submitted another commit accordingly. Closing this conversation